utils.go

```diff
@@ -11,11 +11,6 @@ func ParseIDMapping(UIDMapSlice, GIDMapSlice []string, subUIDMap, subGIDMap stri
 	return types.ParseIDMapping(UIDMapSlice, GIDMapSlice, subUIDMap, subGIDMap)
 }
 
-// GetRootlessRuntimeDir returns the runtime directory when running as non root
-func GetRootlessRuntimeDir(rootlessUID int) (string, error) {
```

Comment:
This is an API break… c/storage, is IIRC, not that strict about them as c/image, I don’t know where exactly the line is.

Reply:
yes it is an API break, just I don't think we should keep around this function that accepts a generic UID.  I would expect the result to be the same as `GetRuntimeDir()` and this is not the case now.

I can move it to a separate PR if it helps, but IMO this variant should just disappear.

Reply:
I really don’t know what the rules for c/storage are.

Personally, it seems to me that we’ve been breaking API similarly… often enough, and I’m not aware of any vocal complaints of wrist slaps; so I think removing this would be reasonably consistent with past practice.

I’m mostly just highlighting this to make it explicit and allow other interested parties to speak up.

Reply:
Also, I don’t think splitting to a separate PR makes much of a difference here: either the API breakage is acceptable, or it isn’t. And if this PR, either way, removed the only callers of the public `GetRootlessRuntimeDir` (and I can see no out-of-repo users in GitHub.com/containers), removing the functions here is trivially small and can be done here.

(By the “2 or 3 separate PRs” above I was thinking:
- A recent version of this: consolidating `GetConfigHome`+`GetRuntimeDir` from Podman etc., keeping `utils.GetRootlessRuntimeDir` mostly unchanged.
- Deprecating `DefaultStoreOptions` (and other APIs that take explicit UID parameters???), and dealing with non-zero fallout of that.
- Then folding `utils.GetRootlessRuntimeDir` into `homedir.GetRuntimeDir`.

I don’t know whether all of that is immediately worthwhile.)